src/main.rs

```diff
@@ -318,6 +329,11 @@ impl Args {
             },
             ("record", Some(submatches)) => {
                 let out_path = output_filename(&std::env::var("HOME")?, submatches.value_of("file"))?;
+                let maybe_duration = match value_t!(submatches, "duration", u64).unwrap_or(0) {
```

Comment:
one last note -- instead of doing `.unwrap_or`, you could match against the option returned by `value_t!` directly (`None => None...`)

Reply:
I had left the `unwrap_or` in place because I kept getting regardless of using `=> { None }` or `=> None` :
```
error[E0308]: mismatched types
   --> src/main.rs:333:21
    |
333 |                     None =>  None ,
    |                     ^^^^ expected enum `std::result::Result`, found enum `std::option::Option`
    |
    = note: expected type `std::result::Result<u64, clap::Error>`
               found type `std::option::Option<_>`
```

Reply:
oops, that's right, this is a Result and not an Option (https://doc.rust-lang.org/std/result/index.html). the right way to match that is

```
match result_thing {
    Err(_) => None,
    Ok(integer) => ...
}
```

Reply:
actually  looking at the [docs for the `value_t` macro](https://docs.rs/clap/2.9.3/clap/macro.value_t!.html) more carefully, it looks like we can actually differentiate between "the argument was missing" and "invalid argument" using `value_t`: you can match against this ErrorKind to differentiate https://kbknapp.github.io/clap-rs/clap/enum.ErrorKind.html

```
Err(ErrorKind::EmptyValue) => None,
Err(x) => return Err(x.into(), // the `x.into()` is an annoying thing we have to do because of Rust
Ok(x) => ...
```